autopilot/contractor.go

```diff
@@ -1620,6 +1606,32 @@ func endHeight(cfg api.AutopilotConfig, currentPeriod uint64) uint64 {
 	return currentPeriod + cfg.Contracts.Period + cfg.Contracts.RenewWindow
 }
 
+func randSelectHosts(candidates map[types.PublicKey]scoredHost, wanted int) (selected []hostdb.Host) {
+	// turn map into slices
+	scores := make([]float64, 0, len(candidates))
```

Comment:
Can we avoid this by updating `randSelectByWeight` to take a slice of `scoredHost`?